src/main/java/ro/unibuc/hello/service/GreetingsService.java

```diff
@@ -0,0 +1,63 @@
+package ro.unibuc.hello.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import ro.unibuc.hello.data.InformationEntity;
+import ro.unibuc.hello.data.InformationRepository;
+import ro.unibuc.hello.dto.Greeting;
+import ro.unibuc.hello.exception.EntityNotFoundException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Component
+public class GreetingsService {
+
+    @Autowired
+    private InformationRepository informationRepository;
+
+    private final AtomicLong counter = new AtomicLong();
+    private static final String helloTemplate = "Hello, %s!";
+    private static final String informationTemplate = "%s : %s!";
+
+    public Greeting hello(String name) {
+        return new Greeting(Long.toString(counter.incrementAndGet()), String.format(helloTemplate, name));
+    }
+
+    public Greeting buildGreetingFromInfo(String title) throws EntityNotFoundException {
+        InformationEntity entity = informationRepository.findByTitle(title);
+        if (entity == null) {
+            throw new EntityNotFoundException(title);
+        }
+        return new Greeting(Long.toString(counter.incrementAndGet()), String.format(informationTemplate, entity.getTitle(), entity.getDescription()));
+    }
+
+    public Greeting saveGreeting(Greeting greeting) {
+        InformationEntity entity = new InformationEntity();
+        entity.setId(greeting.getId());
+        entity.setTitle(greeting.getContent());
+        informationRepository.save(entity);
+        return new Greeting(entity.getId(), entity.getTitle());
+    }
```

Comment:
I find this a bit confusing, we're actually saying we save/update/delete/etc. `Greeting`s, but we actually persist `InformationEntity`es. I mean sure, let's isolate the types we store in the DB from the types we use in the public API, but I think in this case the connection between `Greeting` and `InformationEntity` is not clear.

Reply:
Me too. :)